### DISCUSSION TITLE
Synchronize default Java version on latest runners

### DISCUSSION THREAD

**Discussion by sormuras** [actions/virtual-environments#10509] [Image configuration]
Three different defaults is ... odd.

- [11.0.24+8 (default)](https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md#java) on Ubuntu
- [21.0.4+7.0 (default)](https://github.com/actions/runner-images/blob/main/images/macos/macos-14-Readme.md#java) on macOS
- [8.0.422+5 (default)](https://github.com/actions/runner-images/blob/main/images/windows/Windows2022-Readme.md#java) on Windows

What about setting all to "21"?

**Comment by sormuras** [comment]
Related to:
- https://github.com/actions/runner-images/discussions/10059
- https://github.com/actions/runner-images/discussions/9266

**Comment by erik-bershel** [comment]
Yes, it does look strange, but in fact the main reason is that Java users react extremely negatively to default version updates, which forces us to take such a step most often only when a new OS version is released.
